### PR TITLE
chore(dev) auto-convert Windows line-endings

### PR DESCRIPTION
**Pull request by Tieske**
Git clients on Windows tend to convert line endings to Windows
format, such that Windows editors can be used (is reverted on commit).
This doesn't work if mounting those files from a Unix machine for
development for example (Kong-Vagrant). Lua is agnostic to the format,
but the Unix shell is not. Hence only the scripts in `./kong/bin` are
affected.

This change will have the makefile automatically convert those scripts
back to Unix format. This allows Windows users to work on the Kong code
and custom plugins without other configuration.

### Issues resolved

Fix https://github.com/Kong/kong-vagrant/issues/84

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -61,3 +62,14 @@ pdk-phase-checks:
 	luacov -c t/phase_checks.luacov
 	grep "ngx\\." t/phase_checks.report
 	grep "check_" t/phase_checks.report
+
+win_scripts:
+	@for script in $(WIN_SCRIPTS) ; do \
+	  if [ $$(grep -obUaPc '\015' $$script) -gt 0 ] ; then \
```
Comment (hishamhm) [inline Makefile]: grep is notoriously inconsistent across platforms; the "alphabet soup" of those flags is dangerous (especially if is a dependency of all build targets above).

In particular, I just tested it locally and `-P` doesn't work on my machine. Other distros may have other surprises.

How about making this `win_scripts` unconditional (ie, removing the `if [ $$(grep...` line) and not running it as a dependency of every target (but keeping it in the makefile so it's available there for a quick fixup if needed by Windows users)?

**Comment by hutchic** [comment]
I don't love this change ... majority of Windows development tools are configurable

File: Makefile (hunk 1 above)
Comment (Tieske) [reply Makefile]: That's actually a good idea, I'll update it

File: Makefile [hunk 2]
```diff
@@ -61,3 +62,14 @@ pdk-phase-checks:
 	luacov -c t/phase_checks.luacov
 	grep "ngx\\." t/phase_checks.report
 	grep "check_" t/phase_checks.report
+
+fix-windows:
+	@for script in $(WIN_SCRIPTS) ; do \
+	  if [ $$(grep -obUaPc '\015' $$script) -gt 0 ] ; then \
```
Comment (hishamhm) [inline Makefile]: The `if grep` is still here, this can go away

**Comment by hishamhm** [comment]
Pushed the necessary tweaks myself, squashed and merged. Thanks!

**Comment by Tieske** [comment]
@hishamhm Ahead of me... I just did some minor updates on the github ui, hadn't had a chance to test it. I should have labelled it as WIP, sorry about that.

File: Makefile [hunk 3]
```diff
@@ -61,3 +62,13 @@ pdk-phase-checks:
...
 	grep "ngx\\." t/phase_checks.report
 	grep "check_" t/phase_checks.report
+
+fix-windows:
+	@for script in $(WIN_SCRIPTS) ; do \
+	  echo Converting Windows file $$script ; \
+	  mv $$script $$script.win ; \
+	  tr -d '\015' <$$script.win >$$script ; \
+	  rm $$script.win ; \
+	  chmod 0755 $$script ; \
```
Comment (Tieske) [inline Makefile]: Was this needed btw? I never had an issue without it

Comment (hishamhm) [reply Makefile]: Yes. `>` produces a new file without the executable bit set.

**Pull request merged**